### PR TITLE
Add XML docs for convention classes.

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Part of #214

### CODE REVIEW

File: src/EFCore.Relational/Metadata/Conventions/RelationalMaxIdentifierLengthConvention.cs [hunk 1]
```diff
@@ -8,19 +8,16 @@
...
+    ///     A convention that configures the maximum object identifier length supported by the database. 
     /// </summary>
     public class RelationalMaxIdentifierLengthConvention : IModelInitializedConvention
     {
         /// <summary>
-        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
-        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
-        ///     any release. You should only use it directly in your code with extreme caution and knowing that
-        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        ///     Creates a new instance of <see cref="TableNameFromDbSetConvention" />.
```
Comment (ajcvickers) [inline src/EFCore.Relational/Metadata/Conventions/RelationalMaxIdentifierLengthConvention.cs]: Copy/paste

File: src/EFCore.Relational/Metadata/Conventions/RelationalDbFunctionConvention.cs [hunk 2]
```diff
@@ -12,19 +12,16 @@
...
+    ///     <see cref="DbFunctionAttribute"/>.
     /// </summary>
     public class RelationalDbFunctionConvention : IModelAnnotationChangedConvention
     {
         /// <summary>
-        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
-        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
-        ///     any release. You should only use it directly in your code with extreme caution and knowing that
-        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        ///     Creates a new instance of <see cref="TableNameFromDbSetConvention" />.
```
Comment (ajcvickers) [inline src/EFCore.Relational/Metadata/Conventions/RelationalDbFunctionConvention.cs]: C/P

File: src/EFCore/Update/Internal/UpdateAdapter.cs [hunk 3]
```diff
@@ -62,7 +62,12 @@ public virtual IEnumerable<IUpdateEntry> Entries
         ///     directly from your code. This API may change or be removed in future releases.
         /// </summary>
         public virtual void DetectChanges()
-            => _changeDetector.DetectChanges(_stateManager);
+        {
+            if (_stateManager.Model[ChangeDetector.SkipDetectChangesAnnotation] == null)
```
Comment (ajcvickers) [inline src/EFCore/Update/Internal/UpdateAdapter.cs]: Was this a bug?

Comment (AndriySvyryd) [reply src/EFCore/Update/Internal/UpdateAdapter.cs]: Yes, perf bug

**Pull request merged**